src/spacel/provision/template/app.py

```diff
@@ -74,20 +73,32 @@ def app(self, app, region):
             app_hostname = '%s-%s.%s' % (app.name, orbit.name, orbit.domain)
             params['VirtualHost']['Default'] = app_hostname
 
-        # Expand ELB to all AZs:
-        public_elb_subnets = orbit.public_elb_subnets(region)
-        private_elb_subnets = orbit.private_elb_subnets(region)
-        self._subnet_params(params, 'PublicElb', public_elb_subnets)
-        self._subnet_params(params, 'PrivateElb', private_elb_subnets)
+        if app.loadbalancer == 'true':
+            params['ElbScheme']['Default'] = app.scheme
+            # Expand ELB to all AZs:
+            public_elb_subnets = orbit.public_elb_subnets(region)
+            private_elb_subnets = orbit.private_elb_subnets(region)
+            self._subnet_params(params, 'PublicElb', public_elb_subnets)
+            self._subnet_params(params, 'PrivateElb', private_elb_subnets)
 
-        self._elb_subnets(resources, 'PublicElb', public_elb_subnets)
-        self._elb_subnets(resources, 'PrivateElb', private_elb_subnets)
+            self._elb_subnets(resources, 'PublicElb', public_elb_subnets)
+            self._elb_subnets(resources, 'PrivateElb', private_elb_subnets)
 
         # Expand ASG to all AZs:
-        private_instance_subnets = orbit.private_instance_subnets(region)
-        self._subnet_params(params, 'PrivateInstance', private_instance_subnets)
-        self._asg_subnets(resources, 'PrivateInstance',
-                          private_instance_subnets)
+        if app.availability == 'public':
+            public_instance_subnets = orbit.public_instance_subnets(region)
+            self._subnet_params(params, 'PrivateInstance',
```

Comment:
This feels pretty gnarly: there's ~20 subnet-id's floating around in each VPC, if we start relabelling public subnets as private, troubleshooting from templates becomes hrrrd.

Can we feed _both_ into `_subnet_params`, then only toggle the second argument to `_asg_subnets`?
That way public subnets are still labelled as such, and there's a `{Ref:publicsubnet02}` floating around for sanity if someone's WTF-ing a template. (note: need special handling for `PrivateSubnet01` in this case as `_asg_subnets` just skips it)